packages/repl/src/ReplTable.tsx

```diff
@@ -0,0 +1,95 @@
+import { useState } from "react";
+import type { Results } from "./types";
+
+const tableRowIncrement = 100;
+const maxCellLength = 200;
+
+function cellClass(value: any) {
+  if (value === null) {
+    return "PGliteRepl-null";
+  } else if (typeof value === "number") {
+    return "PGliteRepl-number";
+  } else if (typeof value === "boolean") {
+    return "PGliteRepl-boolean";
+  } else {
+    return "";
+  }
+}
+
+function cellValue(value: any) {
+  let str: string;
+  if (value === null) {
+    str = "null";
+  } else if (typeof value === "number") {
+    str = value.toString();
+  } else if (typeof value === "boolean") {
+    str = value ? "true" : "false";
+  } else if (value instanceof Date) {
+    str = value.toISOString();
+  } else if (Array.isArray(value)) {
+    str = `[${value.map(cellValue).join(", ")}]`;
+  } else if (typeof value === "object") {
+    str = JSON.stringify(value);
+  } else if (ArrayBuffer.isView(value)) {
+    str = `${value.byteLength} bytes`;
+  } else {
+    str = value.toString();
+  }
+  return str.length > maxCellLength ? str.slice(0, maxCellLength) + "…" : str;
+}
+
+export function ReplTable({ result }: { result: Results }) {
+  const [maxRows, setMaxRows] = useState(tableRowIncrement);
+  const rows = result.rows.slice(0, maxRows);
+
+  const showMore = () => {
+    setMaxRows((prev) => prev + tableRowIncrement);
```

Comment:
nit: would we want the `maxRows` to reset upon the `results` changing? Or, if not, in line 79 we might want to bound `maxRows` to `min(maxRows, rows)`